src/provideEndpoints.ts

```diff
@@ -13,15 +14,18 @@ export const defaultApiVersion = 20220511;
 export class EndpointsFactory {
   private readonly environment: Environment;
   private readonly cloudRegion: CloudRegion;
+  private readonly cloudChoice: CloudChoice;
 
   constructor(config?: ServingConfig) {
     this.environment = config?.environment || Environment.PROD;
     this.cloudRegion = config?.cloudRegion || CloudRegion.US;
+    this.cloudChoice = config?.cloudChoice || CloudChoice.GLOBAL_MULTI;
   }
 
   /** Provides the domain based on environment and cloud region. */
   getDomain() {
-    return `https://${this.environment}-cdn.${this.cloudRegion}.yextapis.com`;
+    const cloudChoiceSuffix = this.cloudChoice === CloudChoice.GLOBAL_GCP ? '-gcp' : '';
```

Comment:
So instead of this logic, we could instead define the string values of CloudChoice.GLOBAL_GCP and CloudChoice.GLOBAL_MULTI to be '-gcp' and '', respectively, but I felt that was a little of an unintuitive decision if you look at just that enum file.

Reply:
I think a switch statement here may be more clear but I don't mind it as is